src/python/WMCore/Services/DDM/DDM.py

```diff
@@ -0,0 +1,222 @@
+
+        elif api == 'pollcopy':
+            self.update({'request_id': None,
+                         'item': None,
+                         'site': None,
+                         'status': None,
+                         'user': None})
+
+        elif api == 'cancelcopy':
+            self.update({'request_id': None})
+
+        else:
+            return
+
+    def strip(self):
+        """
+        _strip_
+
+        Method used to drop all the fields with value None. DDM does not
+        handle those quite well, and they better be removed from the request.
+        """
+
+        for key in list(self):
+            if not self[key]:
+                self.pop(key)
+
+    def isEqual(self, ddmReq, exclude=None):
+        """
+        _isEqual_
+
+        Equivalence method.
+        :exclude:
+            a key to be excluded from the comparison.
+        :return:
+            Bool
+        """
+        # first we check the apis - we cannot compare apples to oranges:
+        if not self.api == ddmReq.api:
+            return False
+
+        # in case no key have been excluded it is a normal dict comparison
+        if not exclude:
+            return self == ddmReq
+
+        # The final check: we iterate over the two objects separately in order
+        # to be sure we do not leave any extra key which is present in one of
+        # the objects but not in the other
+        for key in self:
+            if key in exclude:
+                continue
+            try:
+                if ddmReq[key] != self[key]:
+                    return False
+            except KeyError:
+                return False
+
+        for key in ddmReq:
+            if key in exclude:
+                continue
+            try:
+                if ddmReq[key] != self[key]:
+                    return False
+            except KeyError:
+                return False
+        return True
+
+class DDM(Service):
+    """
+    Class which provides client APIs to the DDM service.
+
+    Usage example:
+        api = 'copy'
+        ddmReq = DDMReqTemplate(api)
+        ddm = DDM()
+        ddm.makeRequest(ddmReq, api)
+    """
+
+    def __init__(self, url=None, logger=None, configDict=None):
+        """
+        configDict:
+            Dictionary with parameters that are passed to the super class.
+        """
+        url = url or "https://dynamo.mit.edu"
+        configDict = configDict or {}
+        configDict.setdefault('endpoint', url)
+        configDict.setdefault('cacheduration', 1)  # in hours
+        configDict.setdefault('accept_type', 'application/json')
+        configDict.setdefault('content_type', 'application/json')
+        configDict['logger'] = logger if logger else logging.getLogger()
+        super(DDM, self).__init__(configDict)
+        self['logger'].debug("Initializing DDM with url: %s", self['endpoint'])
+
+    def _getResult(self, ddmReq, apiUrl, callname=""):
+
+        cachedApi = "%s.json" % callname
```

Comment:
I have made this exact same comment few lines bellow in the code itself. Agree, we need to think if we can make benefit of a cache file per DDM instance. Because I want to have one instance of this outside this code and simply feed it with ddmRequstTemplate objects. This was my original idea, but....

Reply:
Looking how PhEDEx wrapper handles this, it calls the `refreshCache` and that one sorts the correct cache file name via `cacheFileName` method. So, I think `cachedApi` can be replaced by `callname`.

BTW, why `callname` has a default value?

Reply:
Done. Remove default value fro `callname`.